std/bigint.d

```diff
@@ -776,6 +809,216 @@ public:
         assertThrown!ConvOverflowException(BigInt("9223372036854775808").to!long);
     }
 
+    /**
+        Implements casting to floating point types.
+     */
+    T opCast(T)() @safe nothrow @nogc const if (isFloatingPoint!T)
+    {
+        return _toFloat!(T, "defaultRounding");
+    }
+
+    // Common code use in opCast!Floating and opCmp!Floating.
+    private T _toFloat(T, string roundingMode)() @safe nothrow @nogc const
+    if (__traits(isFloating, T) && (roundingMode == "defaultRounding" || roundingMode == "truncate"))
+    {
+        import core.bitop : bsr;
+        enum performRounding = roundingMode == "defaultRounding";
+        enum performTruncation = roundingMode == "truncate";
+        static assert(performRounding || performTruncation);
+        enum int totalNeededBits = T.mant_dig + int(performRounding);
+        static if (totalNeededBits <= 64)
+        {
+            // We need to examine the top two 64-bit words, not just the top one,
+            // since the top word could have just a single significant bit.
+            const ulongLength = data.ulongLength;
+            const ulong w1 = data.peekUlong(ulongLength - 1);
+            const ulong w2 = ulongLength < 2 ? 0 : data.peekUlong(ulongLength - 2);
+            const uint w1BitCount = bsr(w1) + 1;
+            ulong sansExponent = (w1 << (64 - w1BitCount)) | (w2 >>> (w1BitCount));
+            size_t exponent = (ulongLength - 1) * 64 + w1BitCount + 1;
+            static if (performRounding)
+            {
+                // If we wanted to round ties away from 0 we could just add
+                // roundUpInc to sansExponent. The more complicated logic is
+                // so ties are rounded to even, which matches the default
+                // rounding behavior when casting int/long to float/double.
```

Comment:
Since this seemingly isn't the default rounding mode on the Win32 & FreeBSD_32 test machines, my inclination is to abandon this rounding mode and round ties away from zero for float/double.